components/survey/SurveyColorContainer.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { cn } from '@/lib/utils';
+import { ChoiceVO, Color } from '@/types/SurveyMessage';
+import HoverWrapper from '../custom/HoverWrapper';
+
+interface SurveyColorContainerProps {
+  choice: ChoiceVO;
+  onClick?: (choice: ChoiceVO) => void;
+  disabled: boolean;
+}
+
+const colorMapper: Record<Color, string> = {
+  red: 'bg-red-600',
+  yellow: 'bg-yellow-200',
+  green: 'bg-green-600',
+  sky: 'bg-sky-400',
+  pink: 'bg-pink-400',
+  purple: 'bg-purple-300',
+  navy: 'bg-indigo-700',
+  white: 'bg-stone-100',
+  black: 'bg-stone-900',
+};
+
+function SurveyColorContainer({
+  choice,
+  onClick,
+  disabled,
+}: SurveyColorContainerProps) {
+  const [isSelect, setIsSelect] = useState(false);
+  return (
+    <HoverWrapper
+      key={choice.id}
+      className={`${cn({
+        disable: disabled && !isSelect,
+      })} ${cn(
+        isSelect ? 'rounded-full border-4 border-violet-700' : '',
+      )} rounded-full border-4`}
```

Comment:
```suggestion
      className={cn(
      'rounded-full border-4',
      {
        disable: disabled && !isSelect,
        'rounded-full border-4 border-violet-700': isSelect
      })}
```

cn 을 쓰기 때문에 class 에 string literal 은 안쓰면 좋을 것 같습니다.